crypto/keys_test.go

```diff
@@ -1,209 +1,41 @@
-		assert.NotEmpty(tt, priv)
-
-		pubKeyBytes, err := PubKeyToBytes(pub)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, pubKeyBytes)
-
-		reconstructedPub, err := BytesToPubKey(pubKeyBytes, P224)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, reconstructedPub)
-		assert.EqualValues(tt, pub, reconstructedPub)
-
-		privKeyBytes, err := PrivKeyToBytes(priv)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, privKeyBytes)
-
-		reconstructedPriv, err := BytesToPrivKey(privKeyBytes, P224)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, reconstructedPriv)
-		assert.EqualValues(tt, priv, reconstructedPriv)
-	})
-
-	t.Run("P256", func(tt *testing.T) {
-		pub, priv, err := GenerateKeyByKeyType(P256)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, pub)
-		assert.NotEmpty(tt, priv)
-
-		pubKeyBytes, err := PubKeyToBytes(pub)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, pubKeyBytes)
-
-		reconstructedPub, err := BytesToPubKey(pubKeyBytes, P256)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, reconstructedPub)
-		assert.EqualValues(tt, pub, reconstructedPub)
-
-		privKeyBytes, err := PrivKeyToBytes(priv)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, privKeyBytes)
-
-		reconstructedPriv, err := BytesToPrivKey(privKeyBytes, P256)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, reconstructedPriv)
-		assert.EqualValues(tt, priv, reconstructedPriv)
-	})
-
-	t.Run("P384", func(tt *testing.T) {
-		pub, priv, err := GenerateKeyByKeyType(P384)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, pub)
-		assert.NotEmpty(tt, priv)
-
-		pubKeyBytes, err := PubKeyToBytes(pub)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, pubKeyBytes)
-
-		reconstructedPub, err := BytesToPubKey(pubKeyBytes, P384)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, reconstructedPub)
-		assert.EqualValues(tt, pub, reconstructedPub)
-
-		privKeyBytes, err := PrivKeyToBytes(priv)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, privKeyBytes)
-
-		reconstructedPriv, err := BytesToPrivKey(privKeyBytes, P384)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, reconstructedPriv)
-		assert.EqualValues(tt, priv, reconstructedPriv)
-	})
-
-	t.Run("P521", func(tt *testing.T) {
-		pub, priv, err := GenerateKeyByKeyType(P521)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, pub)
-		assert.NotEmpty(tt, priv)
-
-		pubKeyBytes, err := PubKeyToBytes(pub)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, pubKeyBytes)
-
-		reconstructedPub, err := BytesToPubKey(pubKeyBytes, P521)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, reconstructedPub)
-		assert.EqualValues(tt, pub, reconstructedPub)
-
-		privKeyBytes, err := PrivKeyToBytes(priv)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, privKeyBytes)
-
-		reconstructedPriv, err := BytesToPrivKey(privKeyBytes, P521)
-		assert.NoError(tt, err)
-		assert.NotEmpty(tt, reconstructedPriv)
-		assert.EqualValues(tt, priv, reconstructedPriv)
-	})
+	for _, keyType := range GetSupportedKeyTypes() {
+		t.Run(string(keyType), func(tt *testing.T) {
```

Comment:
Table driven tests are a go convention.

What do you think about adjusting this from a loop to a table test format? 

https://github.com/golang/go/wiki/TableDrivenTests

Reply:
It seems like for a table driven test it has known input and output. With this unit test it doesn't really have a known output, so the table would only have on dimension an input.

I could explicitly create an array in the test to match this basically or we can keep using GetSupportedKeyTypes

https://github.com/TBD54566975/ssi-sdk/blob/main/crypto/models.go#L45